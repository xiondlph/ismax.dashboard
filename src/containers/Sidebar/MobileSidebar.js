/**
 * Мобильная версия контейнера левой боковой навигационной панели
 */

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { NavLink } from 'react-router-dom'
import {
    DASHBOARD_SIDEBAR_STATE_COLLAPSED
} from '../../constants/Dashboard'

import './MobileSidebar.scss'

export default class MobileSidebar extends Component {

    constructor(props) {
        super(props)

        this.resize = this.resize.bind(this)
    }

    resize(e) {
        if (e.target.innerWidth <= 600 && this.props.sidebar !== DASHBOARD_SIDEBAR_STATE_COLLAPSED) {
            this.props.setMobileSidebarState(DASHBOARD_SIDEBAR_STATE_COLLAPSED)
        }
    }

    componentDidMount() {
        global.addEventListener('resize', this.resize);
    }

    componentWillUnmount() {
        global.removeEventListener('resize', this.resize);
    }

    collapseMobileSidebarState() {
        this.props.setMobileSidebarState(DASHBOARD_SIDEBAR_STATE_COLLAPSED)
    }

    render() {
        const
            { sidebar } = this.props,
            collapsed = sidebar === DASHBOARD_SIDEBAR_STATE_COLLAPSED

        return (
            <div className={classNames({
                'dashboard-sidebar-mobile': true,
                'dashboard-sidebar-mobile_collapsed': collapsed
            })}>
                <div className='dashboard-sidebar-mobile-mask' onClick={::this.collapseMobileSidebarState}/>
                <div className='dashboard-sidebar-mobile-wrap'>
                    <div className='dashboard-sidebar-mobile-menu'>
                        <div className='dashboard-sidebar-mobile-menu-logo'>
                            <span className='dashboard-sidebar-mobile-menu-logo-icon' />
                            ISMAX icsystem
                        </div>
                        <NavLink
                            to='/'
                            activeClassName='dashboard-sidebar-mobile-menu-item_active'
                            exact
                            className='dashboard-sidebar-mobile-menu-item'
                            onClick={::this.collapseMobileSidebarState}
                        >
                            <span className='dashboard-sidebar-mobile-menu-item-icon fa fa-user' />
                            <div className='dashboard-sidebar-mobile-menu-item-text'>Профиль</div>
                        </NavLink>
                        <NavLink
                            to='/payment'
                            activeClassName='dashboard-sidebar-mobile-menu-item_active'
                            className='dashboard-sidebar-mobile-menu-item'
                            onClick={::this.collapseMobileSidebarState}
                        >
                            <span className='dashboard-sidebar-mobile-menu-item-icon fa fa-credit-card' />
                            <div className='dashboard-sidebar-mobile-menu-item-text'>Оплата</div>
                        </NavLink>
                        <NavLink
                            to='/referrals'
                            activeClassName='dashboard-sidebar-mobile-menu-item_active'
                            className='dashboard-sidebar-mobile-menu-item'
                            onClick={::this.collapseMobileSidebarState}
                        >
                            <span className='dashboard-sidebar-mobile-menu-item-icon fa fa-users' />
                            <div className='dashboard-sidebar-mobile-menu-item-text'>Рефералы</div>
                        </NavLink>
                        <div className='dashboard-sidebar-mobile-menu-item'>
                            <span className='dashboard-sidebar-mobile-menu-item-icon fa fa-question' />
                            <span className='dashboard-sidebar-mobile-menu-item-text'>Помощь</span>
                        </div>
                        <div className='dashboard-sidebar-mobile-menu-item'>
                            <span className='dashboard-sidebar-mobile-menu-item-icon fa fa-sign-out' />
                            <span className='dashboard-sidebar-mobile-menu-item-text'>Выход</span>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}

MobileSidebar.propTypes = {
    sidebar: PropTypes.string.isRequired,
    setMobileSidebarState: PropTypes.func.isRequired
}