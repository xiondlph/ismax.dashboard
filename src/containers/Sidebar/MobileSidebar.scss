.dashboard-sidebar-mobile {

  &-wrap {
    background-color: #32404e;
    transition: width 0.5s;
    position: absolute;
    overflow-x: hidden;
    overflow-y: hidden;
    height: 100%;
    z-index: 4;
    width: 0;
    top: 0;

    @include respond-to(mobile) {
      overflow-y: auto;
      width: $sidebar-expanded-width;
    }

    .dashboard-sidebar-mobile_collapsed & {
      overflow-y: hidden;
      width: 0;
    }
  }

  &-mask {
    background-color: #32404e;
    position: fixed;
    display: none;
    height: 100%;
    width: 100%;
    opacity: .3;
    z-index: 3;
    left: 0;
    top: 0;

    @include respond-to(mobile) {
      display: block;
    }

    .dashboard-sidebar-mobile_collapsed & {
      width: 0;
    }
  }

  &-menu {
    padding-top: $sidebar-menu-item-height;
    min-height: calc(100% - 64px);

    &-logo {
      @include box-shadow(0, 1px, 2px, 0, rgba(0, 0, 0, 0.2));
      line-height: $sidebar-menu-item-height;
      height: $sidebar-menu-item-height;
      width: $sidebar-expanded-width;
      background-color: #e05d6f;
      position: absolute;
      font-size: 16px;
      color: #fff;
      z-index: 1;
      top: 0;

      &-icon {
        width: $sidebar-menu-item-height;
        font-family: icsystem;
        display: inline-block;
        text-align: center;
        font-size: 25px;
        float: left;

        &:before {
          content: "\e60c";
        }
      }
    }

    &-item {
      line-height: $sidebar-menu-item-height;
      height: $sidebar-menu-item-height;
      width: $sidebar-expanded-width;
      transition: width 0.5s;
      white-space: nowrap;
      position: relative;
      overflow: hidden;
      font-size: 14px;
      cursor: default;
      display: block;
      color: #adb3b8;

      .dashboard-sidebar_collapsed & {
        width: $sidebar-collapsed-width;

        &:hover {
          overflow: inherit;
        }
      }

      & &-icon {
        line-height: $sidebar-menu-item-height;
        display: inline-block;
        text-align: center;
        margin-left: 5px;
        font-size: 18px;
        width: 59px;
        float: left;
      }

      &-text {
        display: inline-block;
        width: 186px;

        .dashboard-sidebar-menu-item:hover & {
          background-color: #3c4a57;
        }
      }

      &:hover {
        background-color: #3c4a57;
        transition: color 0.5s;
        color: #fff;
      }

      &_active {
        background-color: #475360;

        &:before {
          background-color: #e05d6f;
          position: absolute;
          display: block;
          content: " ";
          height: 100%;
          width: 5px;
          left: 0;
          top: 0;
        }
      }
    }
  }
}