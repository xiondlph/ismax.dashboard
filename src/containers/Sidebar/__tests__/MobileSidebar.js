import React from 'react'
import renderer from 'react-test-renderer'
import { HashRouter } from 'react-router-dom'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import MobileSidebar from '../MobileSidebar'

import {
    DASHBOARD_SIDEBAR_STATE_EXPANDED,
    DASHBOARD_SIDEBAR_STATE_COLLAPSED
} from '../../../constants/Dashboard'

configure({ adapter: new Adapter() })

describe('MobileSidebar', () => {
    test('Проверка MobileSidebar снапшота', () => {

        // Делаем активнам пункт меню payment
        window.location.hash = '#/payment'

        const
            sidebar = DASHBOARD_SIDEBAR_STATE_EXPANDED,
            setMobileSidebarState = jest.fn(),
            component = renderer.create(
                // Нужно обернуть в Router
                <HashRouter>
                    <MobileSidebar sidebar={sidebar} setMobileSidebarState={setMobileSidebarState} />
                </HashRouter>
            )

        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    test('Проверка сворачивания MobileSidebar', () => {
        const
            sidebar = DASHBOARD_SIDEBAR_STATE_EXPANDED,
            setMobileSidebarState = () => {
                mobileSitebar.setProps({ sidebar: DASHBOARD_SIDEBAR_STATE_COLLAPSED })
            },
            mobileSitebar = shallow(
                <MobileSidebar sidebar={sidebar} setMobileSidebarState={setMobileSidebarState} />
            );

        expect(mobileSitebar.hasClass('dashboard-sidebar-mobile')).toEqual(true)
        expect(mobileSitebar.hasClass('dashboard-sidebar-mobile_collapsed')).toEqual(false)

        mobileSitebar.find('.dashboard-sidebar-mobile-mask').first().simulate('click')
        expect(mobileSitebar.hasClass('dashboard-sidebar-mobile_collapsed')).toEqual(true)
    })
})
