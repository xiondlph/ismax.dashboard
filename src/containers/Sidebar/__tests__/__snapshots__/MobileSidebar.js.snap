// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileSidebar Проверка MobileSidebar снапшота 1`] = `
<div
  className="dashboard-sidebar-mobile"
>
  <div
    className="dashboard-sidebar-mobile-mask"
    onClick={[Function]}
  />
  <div
    className="dashboard-sidebar-mobile-wrap"
  >
    <div
      className="dashboard-sidebar-mobile-menu"
    >
      <div
        className="dashboard-sidebar-mobile-menu-logo"
      >
        <span
          className="dashboard-sidebar-mobile-menu-logo-icon"
        />
        ISMAX icsystem
      </div>
      <a
        aria-current={false}
        className="dashboard-sidebar-mobile-menu-item"
        href="#/"
        onClick={[Function]}
        style={undefined}
      >
        <span
          className="dashboard-sidebar-mobile-menu-item-icon fa fa-user"
        />
        <div
          className="dashboard-sidebar-mobile-menu-item-text"
        >
          Профиль
        </div>
      </a>
      <a
        aria-current="true"
        className="dashboard-sidebar-mobile-menu-item dashboard-sidebar-mobile-menu-item_active"
        href="#/payment"
        onClick={[Function]}
        style={Object {}}
      >
        <span
          className="dashboard-sidebar-mobile-menu-item-icon fa fa-credit-card"
        />
        <div
          className="dashboard-sidebar-mobile-menu-item-text"
        >
          Оплата
        </div>
      </a>
      <a
        aria-current={false}
        className="dashboard-sidebar-mobile-menu-item"
        href="#/referrals"
        onClick={[Function]}
        style={undefined}
      >
        <span
          className="dashboard-sidebar-mobile-menu-item-icon fa fa-users"
        />
        <div
          className="dashboard-sidebar-mobile-menu-item-text"
        >
          Рефералы
        </div>
      </a>
      <div
        className="dashboard-sidebar-mobile-menu-item"
      >
        <span
          className="dashboard-sidebar-mobile-menu-item-icon fa fa-question"
        />
        <span
          className="dashboard-sidebar-mobile-menu-item-text"
        >
          Помощь
        </span>
      </div>
      <div
        className="dashboard-sidebar-mobile-menu-item"
      >
        <span
          className="dashboard-sidebar-mobile-menu-item-icon fa fa-sign-out"
        />
        <span
          className="dashboard-sidebar-mobile-menu-item-text"
        >
          Выход
        </span>
      </div>
    </div>
  </div>
</div>
`;
